xcdat/dataset.py

```diff
@@ -315,9 +317,15 @@ def decode_non_cf_time(dataset: xr.Dataset) -> xr.Dataset:
     except ValueError:
         return ds
 
-    ref_date = pd.to_datetime(ref_date)
+    ref_date = pd.to_datetime(ref_date).to_pydatetime()
+
+    if units == "months":
+        data = [ref_date + rd.relativedelta(months=offset) for offset in time.data]
+        data = [cftime.datetime(t.year, t.month, t.day) for t in data]
+    elif units == "years":
+        data = [ref_date + rd.relativedelta(years=offset) for offset in time.data]
+        data = [cftime.datetime(t.year, t.month, t.day) for t in data]
```

Comment:
Is there a way to make the keyword argument vary based on the units (switching between `rd.relativedelta(months=offset)` versus `rd.relativedelta(years=offset)`)